helm/messenger/templates/deployment.yaml

```diff
@@ -82,7 +82,24 @@ spec:
           volumeMounts:
             - name: src
               mountPath: /target/
-    {{- end }}
+        {{- end }}
+        - name: init-conf
+          image: mikefarah/yq:4
+          command: ["sh"]
+          args:
+            - '-c'
+            - >
+              set -ex;
+              touch /tmp/.yaml && printf {{ toYaml $.Values | quote }} >> /tmp/.yaml;
+              for e in {{ join " " $.Values.ingress.hosts | trimPrefix "[" | trimSuffix "]" }}; do yq '. * {"conf":."'$e'" // {}} | withEntries(select(.key == "conf")) | .conf as $root | {} | . = $root' /tmp/.yaml > /tmp/conf/$e.yaml; done;
+              rm -rf /tmp/.yaml;
```

Comment:
I'll try to break down whatever's happening over there:
1) Basically the `yq` command is used to manipulate with YAML files. In order to use the command, we need a file to manipulate to be present in the file system somewhere. `yq` allows input to be env variable, yet personally I find it a bit easier to just `touch` a file and pass it to the command, or otherwise it'll get even bigger and more complex. That's what the `touch` is used for in the first command, it creates the `/tmp/.yaml` file with the `Values` content.
2) Then `sh` iterates over all the Ingress hosts specified in the Helm values and for each one it applies the `yq` transformation. Let's break down the transformation:
- `. * {"conf":."'$e'" // {}}` this will merge the `/tmp/.yaml` file (created in the first step) with the `e` key (e = domain from the Ingress domains list). The `// {}` part will replace possibly `null` output with an empty map, so that if key doesn't exist, the result won't be `null`, it'll just merge the whole `conf` with an empty map.
- `withEntries(select(.key == "conf"))` part simply strips out everything but `conf` key, as the `Values` contains more than just the conf.
- `.conf as $root | {} | . = $root` - these commands make the `conf` key the root of the YAML tree, as otherwise the YAML will be `conf: server: ...` instead of the desired `server: ...`.
3) Then I delete the `/tmp/.yaml` created in the first step. Just in case.